src/main/java/UI.java

```diff
@@ -0,0 +1,107 @@
+import tasklist.Task;
+import java.util.ArrayList;
+
+public class UI {
+    public static void printDuke() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        System.out.print("Hello! I'm Duke\n" + "What can I do for you?\n");
+    }
+
+    public static void bye() {
+        UI.printLine();
+        UI.addSpaces("%sBye. Hope to see you again soon!",5);
+        UI.printLine();
+    }
+
+    public static void printLine() {
+        UI.addSpaces("%s____________________________________________________________",4);
+    }
+
+    public static void printTask(ArrayList<Task> taskList) {
+        UI.printLine();
+        UI.addSpaces("%sGot it. I've added this task:",5);
+        UI.addSpaces("%s" + taskList.get(taskList.size() - 1).toString(),6);
+        UI.addSpaces("%sNow you have " + (taskList.size()) + " tasks in the list.",5);
+        UI.printLine();
```

Comment:
I believe digits 5,6,4 will be deemed as magic numbers here in the codes? Perhaps can store them in constant variables?

Reply:
Is this considered as magic numbers when it is explained in addSpaces() method?

Reply:
```
public static void addSpaces(String text, Integer value) {
        System.out.printf((text) + "%n", " ".repeat(value));
    }
```

Oh, yes I was initially thrown off by the above addSpaces() code with the brackets (text) - so the numbers appear cryptic to me, but I look at the above code usage line 27 - 30 I see now how it is supposed to work.
Kindly disregard my previous comment, thank you!

Reply:
Thanks for confirming it!